### PR TITLE
Cu 3c0r44m setup sockets

### PR DESCRIPTION
**Pull request by MaximeWrlz**

### CODE REVIEW

**Comment by FlavianEng** [comment]
Task linked: [CU-3c0r44m Setup Sockets](https://app.clickup.com/t/3c0r44m)

File: src/engine/gateways/gameGateway.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export class GameGateway implements OnModuleInit {
+  public logger = new Logger();
+
+  @WebSocketServer()
+  public server: Server;
+
+  public onModuleInit() {
+    this.server.on('connection', (socket) => {
+      this.logger.log(socket.id);
+      this.logger.log('Connected');
```
Comment (FlavianEng) [inline src/engine/gateways/gameGateway.ts:15]: Isn't async ?

Comment (FlavianEng) [inline src/engine/gateways/gameGateway.ts:15]:
```suggestion
      this.logger.log(`Socket ${socket.id} connected`);
```

**Review by FlavianEng — DISMISSED** [review]

File: src/engine/gateways/gameGateway.ts (hunk 1 above)
Comment (FlavianEng) [reply src/engine/gateways/gameGateway.ts]: Maxime > No, it's not necessary

File: src/engine/gateways/gameGateway.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Logger, OnModuleInit } from '@nestjs/common';
+import { WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
+import { Server } from 'socket.io';
```
Comment (FlavianEng) [inline src/engine/gateways/gameGateway.ts:3]:
```suggestion
import { Server as SocketServer } from 'socket.io';
```

File: src/engine/gateways/gameGateway.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+@WebSocketGateway({ cors: { origin: '*' } })
+export class GameGateway implements OnModuleInit {
+  public logger = new Logger();
+
+  @WebSocketServer()
+  public server: Server;
+
+  public onModuleInit() {
+    this.server.on('connection', (socket) => {
```
Comment (FlavianEng) [inline src/engine/gateways/gameGateway.ts:13]:
```suggestion
  public socketServer: SocketServer;

  public onModuleInit() {
    this.socketServer.on('connection', (socket) => {
```

**Review by FlavianEng — CHANGES_REQUESTED** [review]

**Review by FlavianEng — APPROVED** [review]

**Review by ManonFargues — APPROVED** [review]

**Pull request merged**